### PR TITLE
Refactor experiment page bundle

### LINKED ISSUE #171: [Chore] Experiment page bundle refactoring
Most of the code that can be found under the experiment page bundle in the `Single Cell Expression Atlas` repository is not belongs to that repository. It should be moved to the `atlas-components` repository by creating 1 or more new components and creating a wrapper around that code in the `single cell` repository and render it.
There are 3 subtask for this story:
- [x] move the code from the `atlas-web-single-cell` repository under `app/src/main/javascript/bundles/experiment-page` folder to the `atlas-components` repository and creating 1 or more new components as it fits
- [x] clean the code: make it easy to maintain, testable, understandable by anyone. I added an example yesterday in Slack. (https://ebi-fg.slack.com/archives/C9P2QETJ9/p1726071369031389)
- [x] creating a wrapper in the `single cell` repository in the `app/src/main/javascript/bundles/experiment-page` folder around the new code and render it

### PR DESCRIPTION
**Pull request by lingyun1010**
We are going to simplify the logic and deployment of experiment page router component and trying to move the bundle to [the front end component repository](https://github.com/ebi-gene-expression-group/atlas-components/tree/chore%2Frefactor-experiment-page-router-component). The package has not been officially published yet, so in this PR, I only installed a local compiled package, you may test the refactoring results by doing the same.

### CODE REVIEW

File: app/src/main/javascript/bundles/experiment-page/package.json [hunk 1]
```diff
@@ -1,43 +1,19 @@
...
-    "@babel/preset-react": "^7.10.4",
-    "babel-loader": "^8.1.0",
-    "clean-webpack-plugin": "^3.0.0",
-    "file-loader": "^6.1.0",
-    "image-webpack-loader": "^7.0.0",
-    "webpack": "^4.44.1",
-    "webpack-cli": "^3.3.12",
-    "webpack-dev-server": "^3.11.0",
-    "whatwg-fetch": "^3.4.1"
+    "@ebi-gene-expression-group/scxa-experiment-page": "^1.0.0-batch",
```
Comment (ke4) [inline app/src/main/javascript/bundles/experiment-page/package.json]: Is the version not `1.0.0-test`?
that's what I saw in your other PR: https://github.com/ebi-gene-expression-group/atlas-components/pull/182/files in the `package.json` file.

File: app/src/main/javascript/bundles/experiment-page/index.js [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
-      species={options.content.species}
-      tabs={options.content.tabs}
-      anatomogram={options.content.anatomogram} />,
+        atlasUrl={options.atlasUrl}
+        resourcesUrl={options.resourcesUrl}
+        experimentAccession={options.content.experimentAccession}
+        accessKey={options.content.accessKey}
+        species={options.content.species}
+        tabs={options.content.tabs}
+        anatomogram={options.content.anatomogram}
```
Comment (ke4) [inline app/src/main/javascript/bundles/experiment-page/index.js]: I can not see `anatomogram` property in the `ExperimentPageRouter` component in the `scxa-experiment-page` package.

**Review by ke4 — CHANGES_REQUESTED** [review]
Thanks for your work.
I added just 2 comments to double check my understanding of the related code in the scxa-experiment-page package.

File: app/src/main/javascript/bundles/experiment-page/package.json (hunk 1 above)
Comment (upendrakumbham) [reply app/src/main/javascript/bundles/experiment-page/package.json]: yes, it is. Now it is` -batch` here.

**Review by upendrakumbham — CHANGES_REQUESTED** [review]
Please add validations where you need before fetching data and consider `isRequired `props validations mandatory please.

File: app/src/main/javascript/bundles/experiment-page/package.json (hunk 1 above)
Comment (lingyun1010) [reply app/src/main/javascript/bundles/experiment-page/package.json]: Yes, it is `1.0.0-batch` . I will push the front end component version commit.
```A new version of the package @ebi-gene-expression-group/scxa-experiment-page (1.0.0-batch) was published at 2024-11-26T16:06:01.205Z from 193.62.194.247.```

**Comment by lingyun1010** [comment]
> Please add validations where you need before fetching data and consider `isRequired `props validations mandatory please.

I think that's not this PR about. That is one of your previous work isn't it? The validation?

File: app/src/main/javascript/bundles/experiment-page/index.js (hunk 2 above)
Comment (lingyun1010) [reply app/src/main/javascript/bundles/experiment-page/index.js]: I don't understand your questions. What do you mean by `property`? Anatomogram is one of the tabs and if you run the webpack server in the `atlas-component`, you may see the content of anatomogram.

Comment (ke4) [reply app/src/main/javascript/bundles/experiment-page/index.js]: In this file: [packages/scxa-experiment-page/src/ExperimentPageRouter.js](https://github.com/ebi-gene-expression-group/atlas-components/blob/5b717676e8e96077b4fbda44f25006fd794dfbac/packages/scxa-experiment-page/src/ExperimentPageRouter.js) I can see all of these properties defined:

```
const TabCommonPropTypes = {
  atlasUrl: PropTypes.string.isRequired,
  experimentAccession: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  accessKey: PropTypes.string,
  resourcesUrl: PropTypes.string
}
...
ExperimentPageRouter.propTypes = {
  ...TabCommonPropTypes,
  tabs: PropTypes.arrayOf(PropTypes.shape({
    type: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    props: PropTypes.object.isRequired
  })).isRequired
}
```

but not `anatomogram`.
Could you tell me whether `ExperimentPageRouter` is using this property?
If not, then why are we setting it?
Thanks

File: app/src/main/javascript/bundles/experiment-page/package.json (hunk 1 above)
Comment (ke4) [reply app/src/main/javascript/bundles/experiment-page/package.json]: Yes, it is now, but it was not when I commented it.

**Review by ke4 — CHANGES_REQUESTED** [review]
Hi,
I am still not sure what is the `anatomogram` property in the `ExperimentPageRouter`.
I added a longer comment regarding to that one.
Could you please clarify it?
Thanks

**Comment by lingyun1010** [comment]
> Hi, I am still not sure what is the `anatomogram` property in the `ExperimentPageRouter`. I added a longer comment regarding to that one. Could you please clarify it? Thanks

Yes, I think you are right, we have `anatomogram` props under variable `tabs`, i.e., `anatomogram` is an entry of `tabs`, but here is a redundant or bug line, as we don't have `anatomogram` props under `content`. I removed this line, please take a further look.

**Review by ke4 — DISMISSED** [review]
LGTM
Thanks for clarifying my suggestion.

**Review by upendrakumbham — DISMISSED** [review]
LGTM!

**Pull request merged**